oak-benchmarks/src/main/java/org/apache/jackrabbit/oak/benchmark/PropertyFullTextTest.java

```diff
@@ -0,0 +1,322 @@
+ *      -jar ~/.m2/repository/org/apache/jackrabbit/oak-run/1.4-SNAPSHOT/oak-run-1.4-SNAPSHOT.jar \
+ *      benchmark --wikipedia enwiki-20150403-pages-articles.xml.bz2 \
+ *      --base ~/tmp/oak/ <Test Extending this class></> Oak-Tar Oak-Mongo
+ * </pre>
+ * <p>
+ * it will run the benchmark for 15 minutes against TarNS and MongoNS.
+ * </p>
+ */
+public class PropertyFullTextTest extends AbstractTest<PropertyFullTextTest.TestContext> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PropertyFullTextTest.class);
+    private WikipediaImport importer;
+    private Thread asyncImporter;
+    private boolean benchmarkCompleted, importerCompleted;
+    Boolean storageEnabled;
+    String currentFixtureName, currentTest;
+
+
+    public String getCurrentFixtureName() {
+        return currentFixtureName;
+    }
+
+    public String getCurrentTest() {
+        return currentTest;
+    }
+
+
+
+
+    public PropertyFullTextTest(final File dump,
+                                final boolean flat,
+                                final boolean doReport,
+                                final Boolean storageEnabled) {
+        this.importer = new WikipediaImport(dump, flat, doReport) {
+
+            @Override
+            protected void pageAdded(String title, String text) {
+                LOG.trace("Setting title: {}", title);
+                lastTitle.set(title);
+            }
+        };
+        this.storageEnabled = storageEnabled;
+        this.currentTest = this.getClass().getSimpleName();
+    }
+
+
+    /**
+     * helper class to initialise the Lucene/Elastic Property index definition
+     */
+    static class FullTextPropertyInitialiser implements RepositoryInitializer {
+        private String name;
+        private Set<String> properties;
+        private String type;
+        private boolean nodeScope;
+        private boolean async;
+        private boolean analyzed;
+
+        public FullTextPropertyInitialiser(@NotNull final String name,
+                                           @NotNull final Set<String> properties,
+                                           @NotNull final String type) {
+            this.name = checkNotNull(name);
+            this.properties = checkNotNull(properties);
+            this.type = checkNotNull(type);
+            this.async = false;
+            this.nodeScope = false;
+            this.analyzed = false;
+        }
+
+        public FullTextPropertyInitialiser async() {
+            this.async = true;
+            return this;
+        }
+
+        public FullTextPropertyInitialiser nodeScope() {
+            this.nodeScope = true;
+            return this;
+        }
+
+        public FullTextPropertyInitialiser analyzed() {
+            this.analyzed = true;
+            return this;
+        }
+
+        private boolean isAlreadyThere(final @NotNull NodeBuilder root) {
+            return checkNotNull(root).hasChildNode(INDEX_DEFINITIONS_NAME) &&
```

Comment:
I would remove the `checkNotNull` since the property is annotated with `@NotNull`